### PR TITLE
make oprf receiver method concurrent

### PR DESCRIPTION
**Pull request by jvsg**

### CODE REVIEW

**Comment by jvsg** [comment]
Changes since the last push

1. Fixed test failure
2. Made  commit message more descriptive of the change

File: internal/oprf/oprf.go [hunk 1]
```diff
@@ -122,19 +123,42 @@ func (ext *OPRF) Receive(choices *cuckoo.Cuckoo, secretKey []byte, rw io.ReadWri
...
 		bitMapLen := util.Pad(ext.m, baseOTCount)
 		pseudorandomEncoding := make([][]byte, bitMapLen)
-		i := 0
-		for ; i < ext.m; i++ {
-			idx := choices.GetBucket(uint64(i))
+
+		var wg sync.WaitGroup
+		wg.Add(maxProcs)
+		for j := 0; j < maxProcs; j++ {
+			go func(j int) {
```
Comment (juanli16) [inline internal/oprf/oprf.go:142]: is there a reason why we shouldn't just spawn these go routines outside?

File: internal/oprf/oprf.go [hunk 2]
```diff
@@ -122,19 +123,42 @@ func (ext *OPRF) Receive(choices *cuckoo.Cuckoo, secretKey []byte, rw io.ReadWri
 	if err != nil {
 		return nil, err
 	}
+
 	var pseudorandomChan = make(chan [][]byte)
+	var maxProcs = runtime.GOMAXPROCS(0)
+
+	var stepSize = ext.m / maxProcs
+	//var leftOver = ext.m - (stepSize * maxProcs)
```
Comment (juanli16) [inline internal/oprf/oprf.go:131]: remove this commented line.

File: internal/oprf/oprf.go [hunk 3]
```diff
@@ -122,19 +123,41 @@ func (ext *OPRF) Receive(choices *cuckoo.Cuckoo, secretKey []byte, rw io.ReadWri
...
-		for ; i < ext.m; i++ {
-			idx := choices.GetBucket(uint64(i))
+
+		var wg sync.WaitGroup
+		wg.Add(maxProcs)
+		for j := 0; j < maxProcs; j++ {
+			go func(j int) {
+				defer wg.Done()
+				for offset := 0; offset < stepSize; offset++ {
+					k := stepSize*j + offset
```
Comment (juanli16) [inline internal/oprf/oprf.go:144]: nitpick, use `step` instead of `k`.

**Review by juanli16 — APPROVED** [review]

**Pull request merged**